test/test_cert_sorter.py

```diff
@@ -99,67 +97,16 @@ def test_unregistered_status(self, mock_update):
         sorter.is_registered = Mock(return_value=False)
         self.assertEqual(UNKNOWN, sorter.get_status(INST_PID_1))
 
-    # Server doesn't support compliance API, or server is unreachable and
-    # we cannot use the cache for some reason.
-    @patch("subscription_manager.cache.InstalledProductsManager.update_check")
-    def test_no_usable_status(self, mock_update):
-        self.status_mgr.load_status = Mock(return_value=None)
-        self.status_mgr.server_status = None
-        sorter = CertSorter()
-        sorter.is_registered = Mock(return_value=True)
-        self.assertEqual(UNKNOWN, sorter.get_status(INST_PID_1))
-
-    # Consumer has been deleted, overall status should be unknown
-    @patch("subscription_manager.cache.InstalledProductsManager.update_check")
-    def test_deleted_consumer_status(self, mock_update):
-        self.status_mgr.load_status = Mock(return_value=None)
-        self.status_mgr.server_status = None
-        sorter = CertSorter()
-        sorter.is_registered = Mock(return_value=True)
-        status_map = subscription_manager.cert_sorter.ComplianceManager.get_status_map()
-        expected = status_map[subscription_manager.cert_sorter.UNKNOWN]
-        self.assertEqual(expected, sorter.get_system_status())
-
     @patch("subscription_manager.cache.InstalledProductsManager.update_check")
     def test_unregistered_system_status(self, mock_update):
         self.status_mgr.load_status = Mock(return_value=None)
         self.status_mgr.server_status = None
         sorter = CertSorter()
         sorter.is_registered = Mock(return_value=False)
         status_map = subscription_manager.cert_sorter.ComplianceManager.get_status_map()
-        expected = status_map[subscription_manager.cert_sorter.UNKNOWN]
+        expected = status_map[subscription_manager.cert_sorter.DISABLED]
```

Comment:
This is strange. On the unregistered system expected status should be `UNKNOWN`.

Reply:
Right. I was too focused on getting the tests to pass to think about what I was doing. The problem was in inheritance -- `sorter.is_registered` mock never called, it all relied on the actual implementation of ComplianceManager. Thanks for catching this.